### PR TITLE
implement pagination for aws rrs and lb

### PR DESCRIPTION
**Pull request by ideahitme**

### CODE REVIEW

File: pkg/aws/client.go [hunk 1]
```diff
@@ -48,25 +48,39 @@ func New(o Options) *Client {
...
+			return nil, err
+		}
 
-	return rsp.ResourceRecordSets, nil
+		if rsp == nil {
+			log.Warnln("Empty response from AWS ListRecordSets API")
+			break
+		}
+
+		records = append(records, rsp.ResourceRecordSets...)
```
Comment (linki) [inline pkg/aws/client.go]: i think you can use [this helper method](https://godoc.org/github.com/aws/aws-sdk-go/service/route53#Route53.ListResourceRecordSetsPages) to do the pagination.

File: pkg/aws/elb.go [hunk 2]
```diff
@@ -14,6 +17,9 @@ type LoadBalancer struct {
 	CanonicalZoneID string
 }
 
+//LoadBalancerExtractor is a func type which is extract Classic and Application Loadbalancer
```
Comment (linki) [inline pkg/aws/elb.go]: typo: `which is extract` => `which extracts` maybe? also explain the use case (e.g. extract from what)

File: pkg/aws/client.go (hunk 1 above)
Comment (ideahitme) [reply pkg/aws/client.go]: 👍

File: pkg/aws/elb.go [hunk 3]
```diff
@@ -52,39 +69,74 @@ func (c *Client) getCanonicalZoneIDs(endpoints []*pkg.Endpoint) (map[string]stri
...
+		loadBalancers := resp.LoadBalancerDescriptions
+		log.Debugf("Page of elbs. Size: %d. Next marker: %v", len(loadBalancers), aws.StringValue(resp.NextMarker))
+		for _, loadbalancer := range loadBalancers {
+			result = append(result, &LoadBalancer{
+				DNSName:         aws.StringValue(loadbalancer.DNSName),
+				CanonicalZoneID: aws.StringValue(loadbalancer.CanonicalHostedZoneNameID),
+			})
+		}
+
+		if aws.StringValue(resp.NextMarker) == "" {
```
Comment (linki) [inline pkg/aws/elb.go]: this pagination helper could get rid of the manual marker stuff https://godoc.org/github.com/aws/aws-sdk-go/service/elb#ELB.DescribeLoadBalancersPages

File: pkg/aws/elb.go [hunk 4]
```diff
@@ -52,39 +69,74 @@ func (c *Client) getCanonicalZoneIDs(endpoints []*pkg.Endpoint) (map[string]stri
...
+			result = append(result, &LoadBalancer{
+				DNSName:         aws.StringValue(loadbalancer.DNSName),
+				CanonicalZoneID: aws.StringValue(loadbalancer.CanonicalHostedZoneId),
+			})
+		}
+
+		if aws.StringValue(resp.NextMarker) == "" {
+			break
+		}
+		params.SetMarker(aws.StringValue(resp.NextMarker))
```
Comment (linki) [inline pkg/aws/elb.go]: https://godoc.org/github.com/aws/aws-sdk-go/service/elbv2#ELBV2.DescribeLoadBalancersPages

File: pkg/aws/elb.go [hunk 5]
```diff
@@ -52,39 +69,74 @@ func (c *Client) getCanonicalZoneIDs(endpoints []*pkg.Endpoint) (map[string]stri
...
 	}
+
 	return result, nil
+
```
Comment (linki) [inline pkg/aws/elb.go]: stray line :)

**Comment by ideahitme** [comment]
@linki thanks for the review, added the changes to use aws-sdk to implement pagination

**Comment by linki** [comment]
👍

**Pull request merged**